x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/common/http/HttpRequestTests.java

```diff
@@ -105,16 +106,11 @@ public void testXContentSerialization() throws Exception {
         }
         if (randomBoolean()) {
             // micros and nanos don't round trip will full precision so exclude them from the test
-            String safeConnectionTimeout = randomValueOtherThanMany(
-                s -> (s.endsWith("micros") || s.endsWith("nanos")),
-                () -> randomTimeValue()
-            );
-            builder.connectionTimeout(TimeValue.parseTimeValue(safeConnectionTimeout, "my.setting"));
+            builder.connectionTimeout(randomTimeValue(0, 1000, TimeUnit.DAYS, TimeUnit.HOURS, TimeUnit.MINUTES, TimeUnit.SECONDS));
```

Comment:
Based on the existing code and comment, it seems we should also include `TimeUnit.MILLISECONDS`

Reply:
Ah thanks, missed that.